### PR TITLE
Verifying Website Domains

### LINKED ISSUE #1360: Verifying Website Domains
### Brief Summary: 

Verifying website domains for originality can be complicated during a time where internet is used constantly for both good and bad. By using Python 3.8.5 Shell and Windows Power Shell within a Windows Operating System environment, determining the credibility of a website domain name can be done. 

### Key Takeaways: 

Retrieve DNS information by coding in Python Shell. Use website API commands within a Windows Power Shell. Identify the complexities of domain modifications.   

### References: 
'N/A'

### PR DESCRIPTION
**Pull request by pkalynan**
### Article Checklist
**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process, quality, and style-once published, you can then submit any number of articles.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?
Images were screenshots of running code on my computer, and quotes were not referring to a specific source they were mostly to emphasize a name of a package, application, or database. Added a resource section to cite documentation.

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)** 

This fixes #1360

### CODE REVIEW

**Comment by pkalynan** [comment]
This fixes #1360

**Comment by adrianmurage** [comment]
@hectorkambow 
kindly run a plagiarism check

**Comment by hectorkambow** [comment]
plagiarism check done

File: articles/verifying-website-domains/index.md [hunk 1]
```diff
@@ -0,0 +1,390 @@
+## Verifying Website Domains
+
+Verifying website domains for originality can be complicated during a time where the internet is constantly used for both good and bad. By using Python Shell and Windows PowerShell within a Windows environment, determining the credibility of a website domain name can be done.
+
+This topic could interest individuals and/or groups who would like to verify domain names for its validity and credibility through coding.
+
+At the end of this tutorial, it is possible to have a good grasp on retrieving DNS (Domain Name System) information by coding in Python Shell, skills developed or refreshed in website API commands within Windows PowerShell, and a tighter hold on identifying the complexities of domain modifications in an ever-changing technical space.
```
Comment (adrianmurage) [inline articles/verifying-website-domains/index.md:7]: This isn't very clear.

Website API commands?
To me, this implies you'll be interacting with a REST API which is not the case for this guide.

**Comment by pkalynan** [comment]
@adrianmurage, Thank you for taking the time to comment and revise.
I will consider your changes as I continue to offer article ideas.

File: articles/verifying-website-domains/index.md (hunk 1 above)
Comment (adrianmurage) [inline articles/verifying-website-domains/index.md:7]:
```suggestion
At the end of this tutorial, you will understand how to retrieve DNS (Domain Name System) information using the Python Shell,  run `curl` in the Windows PowerShell to crawl websites, and how to identify the complexities of domain modifications in an ever-changing technology space.
```

File: articles/verifying-website-domains/index.md [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+
+Windows PowerShell is structured differently. There are subcategorized features to select specific content in a website language. It is essential to select and include `.Content` at the end of the command.
+
+As shown below, brackets are used to call a website with an API to sift through permissible website layers and must begin with `curl`.
+
+Input:
+
+```PowerShell
+(curl host.io/careers.massiveinsights.works).Content
+```
```
Comment (adrianmurage) [inline articles/verifying-website-domains/index.md:235]: This career page does not exist https://host.io/massiveinsights.works

**Review by adrianmurage — CHANGES_REQUESTED** [review]
@pkalynan 
Great article.
A few notes...
1. all headings must be [H3 and above ](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers).
2. I noted you had "--" as a separator. It's unnecessary, kindly avoid it next time.
3. I highlighted some changes you need to make 

@ me when you're done

File: articles/verifying-website-domains/index.md (hunk 2 above)
Comment (pkalynan) [reply articles/verifying-website-domains/index.md:235]: Must have been a typo.
Corrected in the next commit.
It was properly typed in the screenshot and the output.

[http://host.io/careers-massiveinsights.works
](http://host.io/careers-massiveinsights.works
)

I'll be more careful about that in other articles.

File: articles/verifying-website-domains/index.md (hunk 1 above)
Comment (pkalynan) [reply articles/verifying-website-domains/index.md:7]: I can clarify by shortening it.

I could just revise it to:
"At the end of this tutorial, it is possible to complete a complete a reverse DNS (Domain Name System) lookup using Python Shell and Windows PowerShell"

Would that make this portion clearer?

**Comment by pkalynan** [comment]
@adrianmurage 
Made changes. Let me know if it meets expectations/standards. I can make further changes as needed.

**Comment by adrianmurage** [comment]
@hectorkambow 
Could you kindly take a look at the hero image for this article?
I'm wondering if it's appropriate🤔

**Comment by adrianmurage** [comment]
> 
> 
> @adrianmurage
> Made changes. Let me know if it meets expectations/standards. I can make further changes as needed.

@pkalynan this is a great first contribution.
Keep up the good work. : )

**Comment by adrianmurage** [comment]
> 
> ### Article Checklist
> 
> **NOTE: (1st time contributors - submit ONLY one article to get familiar with the process, quality, and style-once published, you can then submit any number of articles.)**
> #### Preliminary Checks - Formatting and Structure
> 
>     * [x]  Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
> 
>     * [x]  Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
> 
>     * [x]  Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
> 
>     * [x]  Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
> 
>     * [x]  Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
> 
>     * [x]  Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
> 
>     * [x]  Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
> 
>     * [x]  Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
> 
>     * [x]  In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
> 
> 
> #### Grammar Checks
> 
>     * [x]  Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com)
> 
>     * [x]  Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
> 
>     * [x]  Have you added sources for quotes and images that aren't yours?
>       Images were screenshots of running code on my computer, and quotes were not referring to a specific source they were mostly to emphasize a name of a package, application, or database. Added a resource section to cite documentation.
> 
> 
> #### Technical Checks
> 
>     * [x]  Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
> 
> 
> ### Contribution Guidelines
> 
> For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
> 
> This fixes #1360

@pkalynan
Note that the hero image should be 600 x 400 pixels as pointed out here.
Also, go through the list and take note of the things pointed out.

**Comment by pkalynan** [comment]
@adrianmurage 
Okay. Thank you.

**Comment by adrianmurage** [comment]
@hectorkambow 
This article is ready for final review.
This is a new contributor and I'm not certain they have an author page, keep that in mind.

**Comment by hectorkambow** [comment]
> @hectorkambow
> This article is ready for final review.
> This is a new contributor and I'm not certain they have an author page, keep that in mind.

When this is the case - please ensure that we explain to the student how to go about including their author page - short bio and headshot image.  (otherwise it is not ready for final) 👍 

@adrianmurage 

@pkalynan  If you could please include a short bio and headshot image to include at the bottom of your article(s). 

See the author folder in the repo as a guide. 
🚀

**Comment by pkalynan** [comment]
Full Name: Priya Kalyanakrishnan

Short Bio:
Priya studied Analytics. She is skilled in other technical fields including programming in object-oriented languages, web coding, machine learning, and statistical coding. Although she may have studied the core basics, she continues to discover more as technology and interrelated areas of interests evolve.

Image Attached.
pk_avatar

**Comment by adrianmurage** [comment]
Hi @pkalynan 



> 
> 
> Full Name: Priya Kalyanakrishnan
> 
> Short Bio:
> Priya studied Analytics. She is skilled in other technical fields including programming in object-oriented languages, web coding, machine learning, and statistical coding. Although she may have studied the core basics, she continues to discover more as technology and interrelated areas of interests evolve.
> 
> Image Attached.
> pk_avatar

Below are instructions on how to add your author information:

Navigate to the authors' folder

image

Inside it create a folder with your name as the folder name as I did here

image

Then inside the folder, you should have two files; your avatar and the index.md file 

image

Your index.md file should then contain the information about you like so:

image

Once you've done this 
@ me to go through it and when I confirm it's all good we can mark this as ready for final review and have your article published!!

**Comment by pkalynan** [comment]
Hi @adrianmurage 
I have followed your instructions.

**Comment by adrianmurage** [comment]
@pkalynan 
Are these changes on a different PR?
I can't see them on this one

**Comment by pkalynan** [comment]
@adrianmurage 
Yes, unfortunately - I could only have my author information on a different PR.

**Comment by adrianmurage** [comment]
@pkalynan 
You can and should have your author information on this PR.
I've created the directory under authors and added the index.md file
Kindly add your avatar to that directory and @ me when it's done.

**Comment by pkalynan** [comment]
@adrianmurage 
Thank you for your patience. I really appreciate it.
You are correct, I should have added these files and folders myself. I really appreciate your help. 
I added the avatar.

**Comment by adrianmurage** [comment]
@pkalynan 
You're welcome. I'm more than happy to help.
@hectorkambow I think this is now ready for final review.

**Comment by hectorkambow** [comment]
Great work and article 🚀  
Looking forward to more great tutorials
@pkalynan  

cc @adrianmurage

**Pull request merged**